src/instruct.jl

```diff
@@ -242,6 +246,38 @@ for G in [:X, :Y, :Z, :S, :T, :Sdag, :Tdag]
                 instruct!(state, g, (locs, ), control_locs, control_bits)
 end
 
+
+# Specialized
+import YaoBase: rot_mat
```

Comment:
specailize this in `YaoBase`? This specialization is more general and is not dispatched on `register`, I think it is OK to put it entirely to `YaoBase`.

Reply:
And remember to use the full name when you trying to overload some function, even this can be longer, it will be very painful to maintain an implicitly overloaded function.

Reply:
Agree, but let's leave it here for compatibility, I don't want to wait for the tagging of YaoBase.

Reply:
Let's merge it first, then add rot_mat to YaoBase. In the next version, it is ok to remove it.